incubator/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSubscribeFactory.java

```diff
@@ -282,52 +346,95 @@ private void onMqttFlush(
             final int reserved = flush.reserved();
 
             assert acknowledge <= sequence;
-            assert sequence >= replySeq;
+            assert sequence >= initialSeq;
+            assert acknowledge >= initialAck;
 
-            replySeq = sequence;
+            initialSeq = sequence;
 
-            assert replyAck <= replySeq;
+            assert initialAck <= initialSeq;
 
             final OctetsFW extension = flush.extension();
             final MqttFlushExFW mqttFlushEx = extension.get(mqttFlushExRO::tryWrap);
 
             assert mqttFlushEx.kind() == MqttFlushExFW.KIND_SUBSCRIBE;
             final MqttSubscribeFlushExFW mqttSubscribeFlushEx = mqttFlushEx.subscribe();
 
-            filters = mqttSubscribeFlushEx.filters();
-            subscriptionIds.clear();
+            currentFilters = mqttSubscribeFlushEx.filters();
+            messagesSubscriptionIds.clear();
+
+            final KafkaFlushExFW kafkaFlushEx = createKafkaFlushEx(currentFilters);
```

Comment:
We aren't really "creating" the extension as it's a flyweight, prefer to do this on-stack please instead of hoisted to a separate method.